src/core/tsi/transport_security.h

```diff
@@ -111,6 +111,9 @@ tsi_result tsi_construct_allocated_string_peer_property(
 tsi_result tsi_construct_string_peer_property_from_cstring(
     const char *name, const char *value, tsi_peer_property *property);
 
+/* Destroy plugin registration funcion. */
+void tsi_register_destroy_plugin(void (*destroy)(void));
```

Comment:
I'm not necessarily opposed to having a separate mechanism for TSI implementations; we do something similar for (e.g.) resolver and LB policy implementations.  But if we're going to do that, then I think we should provide a complete interface, which provides hooks for both initialization and destruction.  For example:

```
void tsi_register_implementation(void (*init)(), void (*destroy)());
```

This way, implementations like GTS can have their own one-time global initialization and destruction code.